### PR TITLE
feat(payment-stripe): promo code, coupon

### PR DESCRIPTION
**Pull request by a-sava**

### CODE REVIEW

File: src/interfaces/payment-stripe/interfaces.ts [hunk 1]
```diff
@@ -4,9 +4,11 @@ type TCurrency = 'usd' | 'eur';
 
 type TBalance = Record<TCurrency, number>;
 
+type TCouponDuration = 'ones' | 'repeating' | 'forever';
```
Comment (OlegDO) [inline src/interfaces/payment-stripe/interfaces.ts]: use enum?

File: src/interfaces/payment-stripe/methods/coupon/create.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import type ICoupon from '../../entities/coupon';
+import type { TCouponDuration, TCurrency } from '../../interfaces';
+
+interface ICreateCouponInput {
+  duration: TCouponDuration;
+  products: string[];
```
Comment (OlegDO) [inline src/interfaces/payment-stripe/methods/coupon/create.ts:8]:
```suggestion
  productIds: string[];
```

**Review by OlegDO — CHANGES_REQUESTED** [review]

File: src/interfaces/payment-stripe/methods/coupon/create.ts (hunk 2 above)
Comment (a-sava) [reply src/interfaces/payment-stripe/methods/coupon/create.ts:8]: endpoint use 'products' property

File: src/constants/payment-stripe/coupon-duration.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+/**
+ * Enum for coupon duration
```
Comment (OlegDO) [inline src/constants/payment-stripe/coupon-duration.ts]:
```suggestion
 * Сoupon duration
```

File: src/endpoints.ts [hunk 4]
```diff
@@ -674,6 +680,34 @@ class Endpoints<
...
+      ),
+      update: this.createHandler<IUpdate<IPromoCode>, IView<IPromoCode>>(
+        'payment-stripe.promo-code.update',
+      ),
+      remove: this.createHandler<IQuery<IPromoCode>, IRemove<IPromoCode>>(
+        'payment-stripe.promo-code.remove',
+      ),
+      count: this.createHandler<IQuery<IPromoCode>, ICount>('payment-stripe.promo-code.count'),
+    },
+    coupon: {
```
Comment (OlegDO) [inline src/endpoints.ts:701]: wrong deps, mout out coupon from pomo code methods

File: src/interfaces/payment-stripe/entities/coupon.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  name?: string | null;
+  amountOff?: number | null;
+  percentOff?: number | null;
+  duration?: CouponDuration;
+  durationInMonths?: number | null;
+  maxRedemptions?: number | null;
+  createdAt?: Date;
+  updatedAt?: Date;
+  products?: IProduct[];
+  promoCodes: IPromoCode[];
```
Comment (OlegDO) [inline src/interfaces/payment-stripe/entities/coupon.ts]:
```suggestion
  promoCodes?: IPromoCode[];
```

File: src/interfaces/payment-stripe/methods/coupon/create.ts (hunk 2 above)
Comment (OlegDO) [reply src/interfaces/payment-stripe/methods/coupon/create.ts:8]: if u provided ids it should be proudct ids, write todo and fix later endpoint input

**Review by OlegDO — APPROVED** [review]

**Pull request merged**